src/lustre/lustreAstDependencies.ml

```diff
@@ -237,36 +434,666 @@ let rec  mk_decl_map: LA.declaration IMap.t -> LA.t -> (LA.declaration IMap.t) g
+       connect_g_pos (List.fold_left (fun g i -> remove g i) rhs_g is)  arr' p
+    (* None of these items below are supported at parsing yet. *)
+    | LA.TupleStructItem (p, _)
+      | LA.TupleSelection (p, _, _)
+      | LA.FieldSelection (p, _, _)
+      | LA.ArraySliceStructItem (p, _, _)
+      ->  Lib.todo ("Parsing not supported" ^ __LOC__ ^ " " ^ Lib.string_of_t Lib.pp_print_position p) in
+  fun m -> function
+        | Equation (pos, (LA.StructDef (p, lhss)), e) ->
+           (* We need to find the mapping of graphs from the 
+              lhs of the equation to the rhs of the equations 
+              such that the width of each of the lhs and rhs 
+              especially if it is just an identifier.
+            *)
+           let rhs_g = mk_graph_expr2 m (LH.abstract_pre_subexpressions e) in
+           (* There are only three possible cases due to the 
+              way we have structures *)
+           (* Case 1: There is only 1 lhss and hence all 
+              elements in rhs depend on that single LHS *)
+           if (List.length lhss = 1)
+           then handle_one_lhs
+                  (List.fold_left
+                     union_dependency_analysis_data
+                     empty_dependency_analysis_data
+                     rhs_g)
+                  (List.hd lhss)
+           (* Case 1: There is only 1 rhs and hence all elements 
+              in lhs depend on that single RHS 
+              I am skeptical about this method, 
+              but we cannot do better unless we do some
+              assignment unfolding, that we cannot at this point. *)
+           else if (List.length rhs_g = 1)
+           then List.fold_left union_dependency_analysis_data empty_dependency_analysis_data
+                  (List.map (handle_one_lhs (List.hd rhs_g)) lhss)
+           (* Case 3: The happy case RHS and LHS should have same number 
+              of items and we can do a one to one mapping*)
+           else List.fold_left union_dependency_analysis_data empty_dependency_analysis_data (List.map2 (handle_one_lhs) rhs_g lhss)
+        | _ -> empty_dependency_analysis_data
+(** Make a dependency graph from the equations. Each LHS has an edge that goes into its RHS definition. *)
+             
+let rec mk_graph_node_items: node_summary -> LA.node_item list -> dependency_analysis_data =
+  fun m -> function
+  | [] -> empty_dependency_analysis_data
+  | (Body eqn) :: items ->
+     union_dependency_analysis_data
+       (mk_graph_eqn m eqn)
+       (mk_graph_node_items m items) 
+  | _ :: items -> mk_graph_node_items m items
+(** Traverse all the node items to make a dependency graph  *)
+
+let analyze_circ_node_equations: node_summary -> LA.ident list -> LA.node_item list -> unit graph_result =
+  fun m consts eqns ->
+  Log.log L_trace "Checking circularity in node equations"
+  ; let ad = mk_graph_node_items m eqns in
+    let dg' = List.fold_left (fun g i -> G.remove_vertex g i) ad.graph_data consts in
```

Comment:
The only call to `analyze_circ_node_equations` receives the empty list as the second argument. Is this line obsolete?

Reply:
removed the extra argument.